### PR TITLE
[netcore] Remove approx. 10 icalls.

### PR DESCRIPTION
**Pull request by jaykrell**
Marshal_PtrToStringUni:
  They are implemented in C#.
  The C# is equivalent to the C, and could be used for classic runtime, except for
   the treatment of  large/negative lengths.

The rest: No details except that not used in netcore.

### CODE REVIEW

File: mono/metadata/marshal.c [hunk 1]
```diff
@@ -5187,6 +5187,8 @@ ves_icall_System_Runtime_InteropServices_Marshal_PtrToStringAnsi_len (const char
 	return mono_string_new_utf8_len (mono_domain_get (), ptr, len, error);
 }
 
+#if !ENABLE_NETCORE
```
Comment (EgorBo) [inline mono/metadata/marshal.c]: Does this #if make sense? these methods won't be linked on netcore and we already have a lot of non-netcore icall impls in this file.

Comment (jaykrell) [reply mono/metadata/marshal.c]: It is true, it is not needed, and has a downside. I’ll remove it.

Comment (lambdageek) [reply mono/metadata/marshal.c]: Without the ifdef, we will get build warnings about non-static functions without a prototype, no?

Comment (vargaz) [reply mono/metadata/marshal.c]: Also, the ifdefs are a good reminder of code which is no longer needed on netcore.

**Comment by jaykrell** [comment]
That’s what I had earlier but Egor objected.

**Comment by jaykrell** [comment]
The advantage of omiting the ifdef is that you don’t have to compile n different times to check if things compile.

To find code to remove, link /verbose (/opt:ref) tells you what it removed. Presumably the other linkers can do that too.

**Comment by jaykrell** [comment]
For the warnings, we can add prototypes, esp. once handles are removed, in which I case I add them anyway. 

Or disable the warning. It isn’t actually useful. Msvc does not issue it.
It is only useful for functions you actually call, and these are not called.

**Comment by jaykrell** [comment]
Maybe a more useful review is to make the PR more complete, based on Egor's suggestion?
`StringToHGlobalUni`?
`StringToHGlobalAnsi`?
`copy_from_unmanaged_fixed`?
`copy_to_unmanaged_fixed`?
`AllocCoTaskMemSize`?
I stopped looking. There could be more.

**Comment by jaykrell** [comment]
Also:
PrelinkAll
PtrToStringAnsi two overloads

for about 10 total.
I updated the PR such that, should be:
 - no warnings about missing prototypes
 - obvious signs of code to be removed if the code bases are forked
 - still compile "all" the functions independent of configuration, so compiling one configuration lends more evidence that the other is not broken. Ultimately it'd be nice if the compile-time ifdef could be runtime, and one runtime could be both compatible with old mono and new netcore, unless it really is easier to maintain otherwise.

File: mono/metadata/icall.c [hunk 2]
```diff
@@ -8589,9 +8589,14 @@ ves_icall_System_Runtime_InteropServices_Marshal_Prelink (MonoReflectionMethodHa
 	prelink_method (MONO_HANDLE_GETVAL (method, method), error);
 }
 
+void
+ves_icall_System_Runtime_InteropServices_Marshal_PrelinkAll (MonoReflectionTypeHandle type, MonoError *error);
```
Comment (vargaz) [inline mono/metadata/icall.c]: Why is this better than putting an ifdef around the function ?

**Comment by jaykrell** [comment]
@monojenkins build failed

File: mono/metadata/icall.c (hunk 2 above)
Comment (jaykrell) [reply mono/metadata/icall.c]: It lets you compile for netcore, but still verify that you didn't break compilation of non-netcore. And vice versa.

**Pull request merged**